toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -967,3 +975,175 @@ func checkEnvironmentVars() error {
+	}
+	defer loopback.Close()
+
+	diskPartitions, err := diskutils.GetDiskPartitions(loopback.DevicePath())
+	if err != nil {
+		return err
+	}
+
+	systemBootPartition, err := findSystemBootPartition(diskPartitions)
+	if err != nil {
+		return err
+	}
+	bootPartition, err := findBootPartitionFromEsp(systemBootPartition, diskPartitions, buildDir)
+	if err != nil {
+		return err
+	}
+
+	systemBootPartitionTmpDir := filepath.Join(buildDir, tmpEspParitionDirName)
+	// Temporarily mount the esp partition.
+	systemBootPartitionMount, err := safemount.NewMount(systemBootPartition.Path, systemBootPartitionTmpDir, systemBootPartition.FileSystemType, 0, "", true)
+	if err != nil {
+		return fmt.Errorf("failed to mount esp partition (%s):\n%w", bootPartition.Path, err)
+	}
+	defer systemBootPartitionMount.Close()
+
+	bootPartitionTmpDir := filepath.Join(buildDir, tmpParitionDirName)
+	// Temporarily mount the boot partition.
+	bootPartitionMount, err := safemount.NewMount(bootPartition.Path, bootPartitionTmpDir, bootPartition.FileSystemType, 0, "", true)
+	if err != nil {
+		return fmt.Errorf("failed to mount partition (%s):\n%w", bootPartition.Path, err)
+	}
+	defer bootPartitionMount.Close()
+
+	// Read the stub path.
+	stubPath := filepath.Join(bootPartitionTmpDir, "linuxx64.efi.stub")
+
+	// Prepare kernel cmdline arguments.
+	grubCfgFullPath := filepath.Join(bootPartitionTmpDir, "grub2/grub.cfg")
+	if err != nil {
+		return fmt.Errorf("failed to stat file (%s):\n%w", grubCfgFullPath, err)
+	}
+
+	// Read the grub.cfg file.
+	grubCfgContent, err := file.Read(grubCfgFullPath)
+	if err != nil {
+		return fmt.Errorf("failed to read file (%s):\n%w", grubCfgFullPath, err)
+	}
+
+	// Split the file content into lines and search for the 'linux' line
+	lines := strings.Split(string(grubCfgContent), "\n")
+	var linuxLine string
+	for _, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmedLine, "linux") {
+			linuxLine = trimmedLine
+			break
+		}
+	}
+	if linuxLine == "" {
+		return fmt.Errorf("failed to find linux line in %s", grubCfgFullPath)
+	}
+
+	// Find all ukify config files under bootPartitionTmpDir.
+	configFiles := []string{}
+	err = filepath.Walk(bootPartitionTmpDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("error accessing path %s: %w", path, err)
+		}
+		if !info.IsDir() && strings.HasPrefix(info.Name(), "ukify_") && strings.HasSuffix(info.Name(), ".conf") {
+			configFiles = append(configFiles, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to find ukify config files: %w", err)
+	}
+
+	// Loop over each config file and build the UKI.
+	for _, configFile := range configFiles {
+		logger.Log.Infof("Processing ukify config: %s", configFile)
+
+		// Read the Linux value from the ukify config.
+		linuxValue, err := retrieveLinuxFromUkifyConf(configFile)
```

Comment:
You should use the ini package here. It should hopefully make the code a lot cleaner.